src/mmap.rs

```diff
@@ -525,6 +560,34 @@ impl<B: NewBitmap> GuestMemoryMmap<B> {
     }
 }
 
+#[cfg(feature = "xen")]
+impl<B: NewBitmap> GuestMemoryMmap<B> {
+    /// Creates a container and allocates anonymous memory for guest memory regions.
+    ///
+    /// Valid memory regions are specified as a sequence of (Address, Size, Option<FileOffset>)
+    /// tuples sorted by Address.
+    pub fn from_ranges_with_files<A, T>(ranges: T) -> result::Result<Self, Error>
+    where
+        A: Borrow<(GuestAddress, usize, Option<FileOffset>, u32, u32)>,
```

Comment:
I find the A a bit confusing here. Is it just to be able to spell out the item separately?

If so, it should also be possible to just leave out the `Item` when constraining `T` and add a line: `T::Item: Borrow<...>`.